packages/core/core/src/Parcel.js

```diff
@@ -270,38 +254,19 @@ export default class Parcel {
       this.#reporterRunner.report({
         type: 'buildStart',
       });
-      let request = createAssetGraphRequest({
-        name: 'Main',
-        entries: options.entries,
+
+      let request = createParcelBuildRequest({
         optionsRef: this.#optionsRef,
-        shouldBuildLazily: options.shouldBuildLazily,
         requestedAssetIds: this.#requestedAssetIds,
-      }); // ? should we create this on every build?
+        signal,
+      });
+
       let {
-        assetGraph,
+        bundleGraph,
+        bundleInfo,
         changedAssets,
         assetRequests,
-      } = await this.#requestTracker.runRequest(request, {
-        force: options.shouldBuildLazily && this.#requestedAssetIds.size > 0,
-      });
-
-      this.#requestedAssetIds.clear();
-
-      let bundleGraphRequest = createBundleGraphRequest({
-        assetGraph,
-        optionsRef: this.#optionsRef,
-      });
-
-      // $FlowFixMe Added in Flow 0.121.0 upgrade in #4381
-      let bundleGraph = await this.#requestTracker.runRequest(
-        bundleGraphRequest,
-      );
-
-      // $FlowFixMe Added in Flow 0.121.0 upgrade in #4381 (Windows only)
-      dumpGraphToGraphViz(bundleGraph._graph, 'BundleGraph');
-
-      await this.#packagerRunner.writeBundles(bundleGraph);
-      assertSignalNotAborted(signal);
+      } = await this.#requestTracker.runRequest(request, {force: true});
```

Comment:
Currently we force the parcel build request to run every time, even when nothing changed. This doesn't cause the sub-requests to re-run – they will return cached results. But if we want to emit a build success event with the bundle graph for builds where nothing actually changed, we need to do this to get the results.

Reply:
Could the the parcle build request do `api.storeResult` instead of only returning it? Not sure if that's more overhead than just rerunning